### PR TITLE
Update s2wasm for 0xc changes

### PR DESCRIPTION
**Pull request by dschuff**
Several updates for s2wasm and its tests:
* Add explicit drops where they are emitted by LLVM already
* Convert loops (which are still modeled in the old way by LLVM) to wrap them in an explicit block (for the exit label). This also allows simplifying the loop creation (no need to post-process the implicit block which is the loop's body). After the engines update to 0xc we should update LLVM to model loops in the 0xc way, but for now it remains compatible with 0xb and 0xc.
* Fix the order of the calls to `setTee()` when creating `tee_local`s
* Add an explicit drop when creating the `_start` entry function wrapper if needed
* Update `dot_s` and `llvm_autogenerated` tests to remove store-results optimization (and few other minor updates)
* Fix the test auto-updater to fail if subprocesses fail

There still seems to be a validation failure when building libc (I think it's from the stricter drop rules, but it may be in the source rather than the compiler), but this at least makes Binaryen's tests pass.

### CODE REVIEW

**Comment by dschuff** [comment]
Forgot to mention the torture tests aren't all passing yet, but this is still a good step.

**Comment by jgravelle-google** [comment]
Seems ok to me, for what it's worth?

File: src/wasm-linker.cpp [hunk 1]
```diff
@@ -209,7 +209,13 @@ void Linker::layout() {
         param->type = type;
         call->operands.push_back(param);
       }
-      block->list.push_back(call);
+      Expression* e = call;
+      if (target->result != none) {
+        auto* drop = out.wasm.allocator.alloc<Drop>();
```
Comment (kripken) [inline src/wasm-linker.cpp]: can use `builder.makeDrop`

Comment (kripken) [inline src/wasm-linker.cpp]: lgtm with that change

**Comment by kripken** [comment]
Is this something you intend to land soon, or keep open for more things?

I ask because I'll start to work on 0xc changes soon. That might cause merge conflicts, I'm not sure.

File: src/wasm-linker.cpp (hunk 1 above)
Comment (dschuff) [inline src/wasm-linker.cpp]: I had it that way to match the rest of that function which didn't use a builder either. But I went ahead and made that whole section use a builder instead.

**Comment by dschuff** [comment]
No, I just got back to fixing the builder thing. I plan on landing it now as soon as CI builds finish.

**Comment by kripken** [comment]
Cool.

**Pull request merged**